dashboard-app/src/App.js

```diff
@@ -1,18 +1,36 @@
 import React, { Component } from 'react';
-import { BrowserRouter as Router, Route } from 'react-router-dom';
 import './App.css';
 import ShowUserList from './components/ShowUserList';
 import PageController from './components/PageController';
 
+export const PageContext = React.createContext({switchPage: () => {}});
+
 class App extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      pageController: null
+    };
+    this.pageContext = {
+      switchPage: (controller) => {
+        this.setState({pageController: controller});
+      }
+    };
+  }
+
   render() {
     return (
-      <Router>
-        <div>
-          <Route exact path='/' component={ShowUserList} />
-          <Route path='/page-controller' component={PageController} />
-        </div>
-      </Router>
+      <div>
+        <PageContext.Provider value={this.pageContext}>
+          {this.state.pageController != null ? (
+            // potentially add info to pagecontroller here
+            <PageController />
+            ) : (
+            <ShowUserList />
```

Comment:
I think you would be adding info to show user list here, page controller would get its info from showUserListClick